easybuild/easyconfigs/g/GROMACS/GROMACS-2024.3-foss-2023b.eb

```diff
@@ -0,0 +1,85 @@
+# This file is an EasyBuild reciPY as per https://github.com/easybuilders/easybuild
+#
+# Copyright:: Copyright 2012-2016 University of Luxembourg / LCSB, Cyprus Institute / CaSToRC,
+#                                 Ghent University / The Francis Crick Institute
+# Authors::
+# * Wiktor Jurkowski <wiktor.jurkowski@gmail.com>
+# * Fotis Georgatos <fotis@cern.ch>
+# * George Tsouloupas <g.tsouloupas@cyi.ac.cy>
+# * Kenneth Hoste <kenneth.hoste@ugent.be>
+# * Adam Huffman <adam.huffman@crick.ac.uk>
+# * Ake Sandgren <ake.sandgren@hpc2n.umu.se>
+# * J. Sassmannshausen <Crick HPC team>
+# * Dugan Witherick <dugan.witherick@warwick.ac.uk>
+# * Christoph Siegert <christoph.siegert@uni-leipzig.de>
+# License::   MIT/GPL
+
+name = 'GROMACS'
+version = '2024.3'
+
+homepage = 'https://www.gromacs.org'
+description = """
+GROMACS is a versatile package to perform molecular dynamics, i.e. simulate the
+Newtonian equations of motion for systems with hundreds to millions of
+particles.
+
+This is a CPU only build, containing both MPI and threadMPI binaries
+for both single and double precision.
+
+It also contains the gmxapi extension for the single precision MPI build.
+"""
+
+toolchain = {'name': 'foss', 'version': '2023b'}
+toolchainopts = {'openmp': True, 'usempi': True}
+
+source_urls = [
+    'https://ftp.gromacs.org/pub/gromacs/',
+    'ftp://ftp.gromacs.org/pub/gromacs/',
+]
+sources = [SOURCELOWER_TAR_GZ]
+patches = [
+    'GROMACS-2023.1_set_omp_num_threads_env_for_ntomp_tests.patch',
+    'GROMACS-2023.1_fix_tests_for_gmx_thread_mpi.patch',
+]
+checksums = [
+    {'gromacs-2024.3.tar.gz': 'bbda056ee59390be7d58d84c13a9ec0d4e3635617adf2eb747034922cba1f029'},
+    {'GROMACS-2023.1_set_omp_num_threads_env_for_ntomp_tests.patch':
+     '7f41bda16c9c2837624265dda4be252f655d1288ddc4486b1a2422af30d5d199'},
+    {'GROMACS-2023.1_fix_tests_for_gmx_thread_mpi.patch':
+     '6df844bb3bbc51180446a3595c61a4ef195e5f975533a04cef76841aa763aec1'},
+]
+
+builddependencies = [
+    ('CMake', '3.27.6'),
+    ('scikit-build-core', '0.9.3'),
+]
+
+dependencies = [
+    ('Python', '3.11.5'),
+    ('SciPy-bundle', '2023.11'),
+    ('networkx', '3.2.1'),
+    ('mpi4py', '3.1.5'),
+]
+
+exts_defaultclass = 'PythonPackage'
+
+exts_default_options = {
+    'source_urls': [PYPI_SOURCE],
+    'use_pip': True,
+    'download_dep_fail': True,
+    'sanity_pip_check': True,
+}
+
+exts_list = [
+    ('gmxapi', '0.4.2', {
```

Comment:
@bedroge This is using an older version of `gmxapi` compared to `GROMACS-2024.1-foss-2023b.eb`, see also https://github.com/easybuilders/easybuild-easyconfigs/pull/20522

Also, we're getting the `gmxapi` from PyPI here, while with GROMACS 2024.1 we used the one included with GROMACS -- we should probably keep doing that?

Reply:
I motivated that change in the PR description (see https://github.com/easybuilders/easybuild-easyconfigs/pull/21430#issue-2530991624). 

Another weird example/behavior of the old approach can be seen in the following inconsistencies (based on the 2024.1 version in EESSI):
```
$ ml GROMACS/2024.1-foss-2023b

$ ls $EBROOTGROMACS/lib/python3.11/site-packages/
gmxapi  gmxapi-0.4.1.dist-info
$ cat $EBROOTGROMACS/lib/python3.11/site-packages/gmxapi-0.4.1.dist-info/METADATA  | grep "^Version"
Version: 0.4.1
$ python -c "from importlib.metadata import version; print(version('gmxapi'))"
0.4.1

$ python -c "import gmxapi; print(gmxapi.__version__)"
0.5.0a1
$ python -c "import gmxapi; print(gmxapi.version.__version__)"
0.5.0a1
```

Reply:
OK, since the installation instructions also recommend installing the latest `gmxapi` version from PyPI (see [here](https://manual.gromacs.org/current/gmxapi/userguide/install.html#install-the-gmxapi-python-package)), we should indeed use this approach.

Thanks for clarifying!